src/libraries/System.ComponentModel.TypeConverter/tests/RegisteredTypesTests.cs

```diff
@@ -0,0 +1,403 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.DotNet.RemoteExecutor;
+using Xunit;
+using static System.ComponentModel.Tests.TypeDescriptorTests;
+
+namespace System.ComponentModel.Tests
+{
+    public class RegisteredTypesTests
+    {
+        private const string TypeDescriptorRequireRegisteredTypesSwitchName = "System.ComponentModel.TypeDescriptor.RequireRegisteredTypes";
```

Comment:
I might have missed this scenario below but asking to make sure - do we have tests for registering a parent type and asking for derived type members?

Reply:
The child properties work the same as they did before - see https://github.com/dotnet/runtime/pull/102094/files#diff-3e07e1115d281208825274551ed27b42fcadbe3832f148953b603a6e696b9d16R212-R213.

However, this does not support calling `TypeDescriptor.GetPropertiesFromRegisteredType<T>()` on a base `T` - you'll get InvalidOperationException. Basically, the previous code is kept as-is w.r.t registering base classes - the properties from base classes are included and trim-safe, but base type themselves are not automatically registered. We could add support to loop through the base classes and register them with the underlying provider, but I'm not sure if this is necessary.

Also, I can update the trimming tests to add a base class to make this support obvious.

Reply:
See the latest commit; I added some tests to clarify this and assert the current behavior.